pkg/operator/targetcontroller/controller.go

```
@@ -94,12 +94,7 @@ func (c *TargetController) sync() error {
 	default:
 	}
 
-	forceRequeue, err := c.syncKubeStorageVersionMigrator(spec, status, objectMetaGeneration)
-	if forceRequeue && err != nil {
```

Wouldn't just changing `&&` to `||` fix the problem?

---

yes, i can try that instead but will also need to bubble up the errors from syncKubeStorageVersionMigrator - as/is  forceRequeue is only true when err is nil, and also, the error from syncKubeStorageVersionMigrator that happens when, for example, the operand ns is deleted, does not get returned, it's buried in the operator status message, while the fn returns nil - so errors like that will never trigger a requeue - that's what I've been observing when I delete the operand ns, the operator gets stuck, only recovers when i refresh the operator pod.

---

updated, ptal

---

this looks wrong. Instead we should return the error if non-nil.

---

agreed, updated, thanks